x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.idp.saml.sp;
+
+import org.joda.time.ReadableDuration;
+
+import java.net.URI;
+
+/**
+ * SAML 2.0 configuration information about a specific service provider
+ */
+public interface SamlServiceProvider {
+    String getEntityId();
+
+    URI getAssertionConsumerService();
+
+    ReadableDuration getAuthnExpiry();
```

Comment:
Since this is just an interface, how we resolve the duration per SP isn't dictated, just that the Response builder doesn't assume it's a global value in the IdP.

For things that _could_ reasonably be set per SP (that is, there's no technical reason why it should be fixed for the IdP), I'd prefer to provide them to on the SP so that the Builder doesn't make assumptions.

One reason I can think of for having a per-SP duration is if the SP doesn't properly handle clock skew so you need to increase the expiry time specifically for that SP.